### PR TITLE
ENH: add "schema" kwarg to io.sql.get_schema method

### LINKED ISSUE #28486: Add support for schema parameter when calling io.sql.get_schema
Currently, there is no way to specify the schema when calling pandas.io.sql.get_schema, the schema is a helpful parameter when working with certain databases such as MSSQL.

```python
import pandas.io.sql

# current state
table_name = 'my_new_table'
sql_create_statement = pandas.io.sql.get_schema(df, table_name, con)
# create statement looks like
"""
CREATE TABLE [my_new_table] (
    [column1] VARCHAR(max) NULL,
    ....
)
"""
```

this adds an extra step to have to parse the file in order to add the schema, otherwise when I execute the script later, the table is created in user default schema (dbo in most cases). This issue can be easily addressed by adding  `schema` parameter to `_create_sql_schema` since `SQLTable` already supports `schema`

```python

import pandas.io.sql

# future state
table_name = 'my_new_table'
schema = 'pypi'
sql_create_statement = pandas.io.sql.get_schema(df, table_name, schema=schema)

# create statement will look like this for MSSQL if con parameter is passed
"""
CREATE TABLE pypi.[my_new_table] (
    [Column1],
    ....
)
"""
```

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes  #28486
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Picking up #33278

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. @jorisvandenbossche if you can give a once over

**Comment by jreback** [comment]
cc @jorisvandenbossche

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -260,6 +260,7 @@ Other enhancements
 - Added :meth:`~DataFrame.set_flags` for setting table-wide flags on a Series or DataFrame (:issue:`28394`)
 - :meth:`DataFrame.applymap` now supports ``na_action`` (:issue:`23803`)
 - :class:`Index` with object dtype supports division and multiplication (:issue:`34160`)
+- :meth:`get_schema` now allows a schema kwarg that will add a schema into the create table statement (:issue:`28486`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.2.0.rst:263]: Mind specifying `io.sql.get_schema` directly?

**Review by mroeschke — COMMENTED** [review]
One minor comment otherwise LGTM

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (arw2019) [reply doc/source/whatsnew/v1.2.0.rst]: Sure! Done

**Review by mroeschke — APPROVED** [review]

File: pandas/io/sql.py [hunk 2]
```diff
@@ -1873,4 +1896,6 @@ def get_schema(frame, name, keys=None, con=None, dtype=None):
 
     """
     pandas_sql = pandasSQL_builder(con=con)
-    return pandas_sql._create_sql_schema(frame, name, keys=keys, dtype=dtype)
+    return pandas_sql._create_sql_schema(
```
Comment (jreback) [inline pandas/io/sql.py:1899]: oh you need to update doc-string and add versionadded 1.2

**Review by jreback — CHANGES_REQUESTED** [review]
ping on green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green

**Comment by arw2019** [comment]
Green

**Comment by mroeschke** [comment]
Thanks @arw2019

**Pull request merged**